File: Array/huyang/有效的括号.kt
class Solution {
fun isValid(s: String): Boolean {
        val stack = LinkedList<Char>()
        for (c in s) {
            when (c) {
                '(' -> stack.push(c)
                '[' -> stack.push(c)
                '{' -> stack.push(c)
                ')' -> if (stack.isNotEmpty() && stack.first == '(') stack.pop() else return false
                ']' -> if (stack.isNotEmpty() && stack.first == '[') stack.pop() else return false
                '}' -> if (stack.isNotEmpty() && stack.first == '{') stack.pop() else return false
            }
        }
        return stack.isEmpty()
    }
}

File: Array/huyang/07之前/有效的括号.cpp
class Solution {
public:
    bool isValid(string s) {
         stack<char> parentheses;
         for(int i=0 ; i < s.size(); i++){
             if(s[i] == '('||s[i] == '{' || s[i] == '['){
                 parentheses.push(s[i]);
             }else{
                 if (parentheses.empty()) return false;
                if(s[i] == ')' && parentheses.top() != '('){
                    return false;
                }
                if(s[i] == '}' && parentheses.top() != '{'){
                    return false;
                }
                if(s[i] == ']' && parentheses.top() != '['){
                    return false;
                }
                parentheses.pop(); 
             }
         }
         return parentheses.empty();
    }
};